### PR TITLE
tune testing support

### PR DESCRIPTION
**Pull request by doudou**
This controls ROCK_TEST_ENABLED (from the Rock cmake macros) based on whether
the underlying package has tests enabled or not, and makes sure that we build
orocos.rb's test oroGen components during the build phase if tests are enabled
for it (and that we do so with the appropriate -j)

### CODE REVIEW

**Comment by doudou** [comment]
Ping @rock-core/core

**Comment by doudou** [comment]
Ok ... I can see that unit testing is very high in the priority lists.

File: 04stable.autobuild [hunk 1]
```diff
@@ -120,10 +120,16 @@ in_flavor 'master', 'stable' do
     end
 
     ruby_package  'tools/orocos.rb' do |pkg|
-        def pkg.update_environment
+        def pkg.invoke_rake(target_name = rake_setup_task)
             super
-            Autobuild.update_environment srcdir
-            Autoproj.env_set 'ORBInitRef',"NameService=corbaname::127.0.0.1"
```
Comment (goldhoorn) [inline 04stable.autobuild]: How is this removal related to test support?

Comment (doudou) [inline 04stable.autobuild]: It's not ... updated

**Comment by goldhoorn** [comment]
As far i understood this enables the build of all test-based extensions by default.
Furthermore this is done also for stable and not only for master.

I would prefer to have a kind of ENV or Autoproj setting that controls if the test suites are build by default instead of activating them automatically.

I have also questions regarding testing in rock in general:

this PR will enable the compilation of all test parts by default
autoproj test will run the tests?
why call autoproj osdeps after autoproj test?

**Comment by goldhoorn** [comment]
Ah okay get it. ```autoproj test --enable``` enables the test_utility for each package. 
The tests got therefore not build by default. Only if this is set by autoproj configuration
